django/tests/otto/test_cleanup.py

```diff
@@ -300,3 +305,91 @@ def test_delete_empty_chats_task(client, all_apps_user):
+        codename="add_userrequest", content_type=content_type_user_request
+    )
+    user.user_permissions.add(permission_user_request)
+
+    user_request = UserRequest.objects.create(
+        access_key=access_key, name="Test Request"
+    )
+
+    # Grant the permissions to the user for OutputFile
+    content_type_output_file = ContentType.objects.get_for_model(OutputFile)
+    permission_output_file_add = Permission.objects.get(
+        codename="add_outputfile", content_type=content_type_output_file
+    )
+    permission_output_file_delete = Permission.objects.get(
+        codename="delete_outputfile", content_type=content_type_output_file
+    )
+    user.user_permissions.add(permission_output_file_add)
+    user.user_permissions.add(permission_output_file_delete)
+    # Verify current time
+    current_time = timezone.now()
+    print(f"Current time: {current_time}")
+    # Create an OutputFile
+    this_file_path = os.path.abspath(__file__)
+    with open(this_file_path, "rb") as f:
+        content = f.read()
+
+    output_file1 = OutputFile.objects.create(
+        access_key=access_key,
+        user_request=user_request,
+        file=ContentFile(content, name="test_file1.txt"),
+        file_name="test_file1.txt",
+    )
+    # Set the creation time to 40 hours ago
+    output_file1.created_at = current_time - timezone.timedelta(hours=40)
+    output_file1.save(access_key=access_key)
+    print(f"Output file 1 created_at: {output_file1.created_at}")
+    # output_file2 created 1 hour ago
+    output_file2 = OutputFile.objects.create(
+        access_key=access_key,
+        user_request=user_request,
+        file=ContentFile(content, name="test_file2.txt"),
+        file_name="test_file2.txt",
+    )
+
+    # Set the creation time to 5 min ago
+    output_file2.created_at = current_time - timezone.timedelta(minutes=5)
+    output_file2.save(access_key=access_key)
+
+    # Run the delete_text_extractor_files task
+    if settings.IS_RUNNING_IN_GITHUB:
+        # No Redis, so we test the code directly
+        call_command("delete_text_extractor_files")
+    else:
+        # Test the task
+        from otto.tasks import delete_text_extractor_files
+
+        delete_text_extractor_files()
+
+    # Check that the old OutputFile is gone
+    old_output_files = OutputFile.objects.filter(
+        access_key=access_key, file_name="test_file1.txt"
+    )
+    assert old_output_files.count() == 0
+
+    # # # Check that the new OutputFile still exists
+    new_output_files = OutputFile.objects.filter(
+        access_key=access_key, file_name="test_file2.txt"
+    )
+    assert new_output_files.count() == 1
+
+    # Check media directory
+    media_folder = os.path.join(settings.MEDIA_ROOT, "ocr_output_files")
+    print(f"Files in media folder: {os.listdir(media_folder)}")
+    assert "test_file1.txt" not in os.listdir(media_folder)
+    assert "test_file2.txt" in os.listdir(media_folder)
```

Comment:
2 test files created, one is 40 hours old and the second is 5 minutes old.
checks:
-First file should be deleted as it is 40 hours old
-Second file should be kept as it is 5 minutes old

Reply:
Permissions had to be explicitly given to create OutputFile objects